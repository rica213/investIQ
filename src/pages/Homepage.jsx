import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import bgHeadline from '../assets/images/bg-headline.jpg';
import Companies from '../components/Companies';
import Search from '../components/Search';
import Summary from '../components/Summary';
import styles from '../styles/Homepage.module.css';
import Modal from '../components/Modal';
import { getAllCompanies } from '../redux/companies/companiesSlice';
import ScrollToTopButton from '../components/ScrollToTopButton';

const Homepage = () => {
  const { isOpen } = useSelector((store) => store.modal);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getAllCompanies());
  }, [dispatch]);
  return (
    <>
      <div
        className={styles.headline}
        style={{ backgroundImage: `url(${bgHeadline})` }}
      >
        <a
          href="https://www.freepik.com/free-vector/investment-data-concept-illustration_12704375.htm#query=finance%20illustration&position=4&from_view=search&track=ais"
          className={styles.attribution}
        >
          Image by storyset
        </a>
        {' '}
        on Freepik
      </div>
      <Search />
      <Summary title="Publicly Traded Companies based in US" />
      <Companies />
      {isOpen && <Modal />}
      <ScrollToTopButton />
    </>
  );
};

export default Homepage;
