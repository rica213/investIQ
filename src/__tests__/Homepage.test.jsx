import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Homepage from '../pages/Homepage';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Homepage component', () => {
  let initialState;
  let store;
  beforeEach(() => {
    initialState = {
      modal: { isOpen: false },
      companies: {
        companies: [
          {
            symbol: 'AAPL',
            companyName: 'Apple Inc.',
            marketCap: 2000000000000,
          },
          {
            symbol: 'MSFT',
            companyName: 'Microsoft Corporation',
            marketCap: 1500000000000,
          },
        ],
        filteredCompanies: [
          {
            symbol: 'AAPL',
            companyName: 'Apple Inc.',
            marketCap: 2000000000000,
          },
          {
            symbol: 'MSFT',
            companyName: 'Microsoft Corporation',
            marketCap: 1500000000000,
          },
        ],
        isLoading: false,
      },
    };
    store = mockStore(initialState);
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Homepage />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with mock data', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Router>
          <Homepage />
        </Router>
      </Provider>,
    );

    expect(
      getByText('Publicly Traded Companies based in US'),
    ).toBeInTheDocument();
    expect(getByText('AAPL')).toBeInTheDocument();
    expect(getByText('MSFT')).toBeInTheDocument();
  });
});
