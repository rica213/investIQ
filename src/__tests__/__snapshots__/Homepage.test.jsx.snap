// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > should match the snapshot 1`] = `
[
  <div
    className="_headline_524292"
    style={
      {
        "backgroundImage": "url(/src/assets/images/bg-headline.jpg)",
      }
    }
  >
    <a
      className="_attribution_524292"
      href="https://www.freepik.com/free-vector/investment-data-concept-illustration_12704375.htm#query=finance%20illustration&position=4&from_view=search&track=ais"
    >
      Image by storyset
    </a>
     
    on Freepik
  </div>,
  <form
    className="_searchForm_ef0f0f"
    onChange={[Function]}
  >
    <input
      className="_searchBy_ef0f0f"
      onChange={[Function]}
      placeholder="Search by Symbol | eg. AAPL"
      type="text"
    />
    <button
      className="_searchBtn_ef0f0f"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </form>,
  <article
    className="_summary_78e985"
  >
    <p>
      Publicly Traded Companies based in US
    </p>
  </article>,
  <ul
    className="_companiesList_67bdf6"
  >
    <a
      className="_link_67bdf6"
      href="/profile"
      onClick={[Function]}
    >
      <li
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <div
          className="_bottom_67bdf6"
        >
          <h4
            className="_name_67bdf6"
          >
            AAPL
          </h4>
          <h5
            className="_symbol_67bdf6"
          />
          <h5
            className="_symbol_67bdf6"
          />
        </div>
      </li>
    </a>
    <a
      className="_link_67bdf6"
      href="/profile"
      onClick={[Function]}
    >
      <li
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <div
          className="_bottom_67bdf6"
        >
          <h4
            className="_name_67bdf6"
          >
            MSFT
          </h4>
          <h5
            className="_symbol_67bdf6"
          />
          <h5
            className="_symbol_67bdf6"
          />
        </div>
      </li>
    </a>
  </ul>,
  <button
    className="_scrollToTop_fd4189"
    onClick={[Function]}
    style={
      {
        "display": "none",
      }
    }
  >
    Top
  </button>,
]
`;
